tailscale/resource_posture_integration.go

```diff
@@ -0,0 +1,145 @@
+package tailscale
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	tsclient "github.com/tailscale/tailscale-client-go/v2"
+)
+
+func resourcePostureIntegration() *schema.Resource {
+	return &schema.Resource{
+		Description:   "The posture_integration resource allows you to manage integrations with device posture data providers. See https://tailscale.com/kb/1288/device-posture for more information.",
+		ReadContext:   resourcePostureIntegrationRead,
+		CreateContext: resourcePostureIntegrationCreate,
+		UpdateContext: resourcePostureIntegrationUpdate,
+		DeleteContext: resourcePostureIntegrationDelete,
+		Schema: map[string]*schema.Schema{
+			"posture_provider": {
+				Type:        schema.TypeString,
+				Description: "The type of posture integration data provider.",
+				Required:    true,
+				ForceNew:    true,
+				ValidateFunc: validation.StringInSlice(
+					[]string{
+						string(tsclient.PostureIntegrationProviderFalcon),
+						string(tsclient.PostureIntegrationProviderIntune),
+						string(tsclient.PostureIntegrationProviderJamfPro),
+						string(tsclient.PostureIntegrationProviderKandji),
+						string(tsclient.PostureIntegrationProviderKolide),
+						string(tsclient.PostureIntegrationProviderSentinelOne),
+					},
+					false,
+				),
+			},
+			"cloud_id": {
+				Type:        schema.TypeString,
+				Description: "Identifies which of the provider's clouds to integrate with.",
+				Optional:    true,
+			},
+			"client_id": {
+				Type:        schema.TypeString,
+				Description: "Unique identifier for your client.",
+				Optional:    true,
+			},
+			"tenant_id": {
+				Type:        schema.TypeString,
+				Description: "The Microsoft Intune directory (tenant) ID. For other providers, this is left blank.",
+				Optional:    true,
+			},
+			"client_secret": {
+				Type:        schema.TypeString,
+				Description: "The secret (auth key, token, etc.) used to authenticate with the provider.",
+				Required:    true,
+			},
+		},
+	}
+}
+
+func resourcePostureIntegrationRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(*Clients).V2
+
+	integrations, err := client.DevicePosture().ListIntegrations(ctx)
```

Comment:
Forgot that I had already filed https://github.com/tailscale/corp/issues/21632 but might be worth implementing a `GET /posture/integration/{integrationId}` at some point and using it here.